api/solarperformanceinsight_api/models.py

```diff
@@ -548,3 +549,291 @@ class UserInfo(StoredObject):
+    predicted_expected = "predicted and expected performance"
+    expected_actual = "expected and actual performance"
+
+
+class ComparePerformanceJob(BaseModel):
+    """Compare one type of performance to another"""
+
+    compare: CompareEnum
+    performance_granularity: PerformanceGranularityEnum
+    _weather_types: Tuple[JobDataTypeEnum, ...] = PrivateAttr()
+    _performance_types: Tuple[JobDataTypeEnum, ...] = PrivateAttr()
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        if self.compare == CompareEnum.predicted_actual:
+            self._weather_types = (
+                JobDataTypeEnum.original_weather,
+                JobDataTypeEnum.actual_weather,
+            )
+            self._performance_types = (JobDataTypeEnum.actual_performance,)
+        elif self.compare == CompareEnum.predicted_expected:
+            self._weather_types = (
+                JobDataTypeEnum.original_weather,
+                JobDataTypeEnum.actual_weather,
+            )
+            self._performance_types = (JobDataTypeEnum.predicted_performance,)
+        elif self.compare == CompareEnum.expected_actual:
+            self._weather_types = (JobDataTypeEnum.actual_weather,)
+            self._performance_types = (JobDataTypeEnum.actual_performance,)
+
+
+JOB_PARAMS_EXAMPLE = dict(
+    system_id=SYSTEM_ID,
+    job_type=dict(
+        compare="predicted and actual performance",
+        performance_granularity="inverter",
+    ),
+    time_parameters=dict(
+        start="2020-01-01T00:00:00+00:00",
+        end="2020-12-31T23:59:59+00:00",
+        step="15:00",
+        timezone="UTC",
+    ),
+    weather_granularity="array",
+    irradiance_type="poa",
+    temperature_type="module",
+)
+
+
+class JobParameters(BaseModel):
+    system_id: UUID
+    job_type: Union[CalculatePerformanceJob, ComparePerformanceJob] = Field(
+        ..., description="Calculation or comparison to be performed"
+    )
+    time_parameters: JobTimeindex
+    weather_granularity: WeatherGranularityEnum
+    # in principle, these both could be on a per model chain/inverter basis,
+    # but easier to just keep everything the same for the system
+    irradiance_type: IrradianceTypeEnum
+    temperature_type: TemperatureTypeEnum
+
+    class Config:
+        schema_extra = {"example": JOB_PARAMS_EXAMPLE}
+
+
+JOB_DATE_META_EXAMPLE = dict(
+    schema_path="/inverters/0",
+    type="actual weather data",
+    filename="inverter_0_weather.csv",
+    data_format="application/vnd.apache.arrow.file",
+    present=True,
+)
+
+
+class JobDataItem(BaseModel):
+    schema_path: str = Field(
+        ..., description="Relative to PV system definition, i.e. /inverters/0/arrays/0"
+    )
+    type: JobDataTypeEnum
+
+
+class JobDataMetadata(JobDataItem):
+    filename: Optional[str] = Field(..., description="Filename of file uploaded")
+    data_format: Optional[str]
+    present: bool = Field(False, description="If the data has been uploaded or not")
```

Comment:
```suggestion
    present: bool = Field(False, description="If the data has been uploaded or not")
    
    class Config:
        schema_extra = {"example": JOB_DATA_META_EXAMPLE}
```
I think with the example above this got left out?